fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -6878,15 +6877,15 @@ private Money[] fetchInterestFeeAndPenaltyTillDate(final LocalDate paymentDate,
             final LoanRepaymentScheduleInstallment installment, boolean isFirstNormalInstallment) {
         Money penaltyForCurrentPeriod = Money.zero(getCurrency());
         Money penaltyAccoutedForCurrentPeriod = Money.zero(getCurrency());
-        Money feeForCurrentPeriod = Money.zero(getCurrency());
-        Money feeAccountedForCurrentPeriod = Money.zero(getCurrency());
-        Money interestForCurrentPeriod = Money.zero(getCurrency());
-        Money interestAccountedForCurrentPeriod = Money.zero(getCurrency());
         int totalPeriodDays = Math.toIntExact(ChronoUnit.DAYS.between(installment.getFromDate(), installment.getDueDate()));
         int tillDays = Math.toIntExact(ChronoUnit.DAYS.between(installment.getFromDate(), paymentDate));
-        interestForCurrentPeriod = Money.of(getCurrency(), BigDecimal
+        Money interestForCurrentPeriod = Money.of(getCurrency(), BigDecimal
                 .valueOf(calculateInterestForDays(totalPeriodDays, installment.getInterestCharged(getCurrency()).getAmount(), tillDays)));
-        interestAccountedForCurrentPeriod = installment.getInterestWaived(getCurrency()).plus(installment.getInterestPaid(getCurrency()));
+        Money interestAccountedForCurrentPeriod = installment.getInterestWaived(getCurrency())
+                .plus(installment.getInterestPaid(getCurrency()));
+        Money feeForCurrentPeriod = installment.getFeeChargesCharged(getCurrency());
+        Money feeAccountedForCurrentPeriod = installment.getFeeChargesWaived(getCurrency())
+                .plus(installment.getFeeChargesPaid(getCurrency()));
         for (LoanCharge loanCharge : this.charges) {
```

Comment:
@fiter-julius-oketayot You are already calculating the feeForCurrentPeriod and feeAccountedForCurrentPeriod before this for loop. Will this loop not double the fee and penalty amounts?

Reply:
@faheem205 the idea is to account for all the fees in a repayment period even if the borrower enters that period by one day. 
For example if fromDate 1st/July, and dueDate is 31st/July and the payment date if 2nd/July
The borrower will have to pay for all the fees for that period

Reply:
This was an issue with how the charge was setup and a small fix has been made to rectify. Thank you @fiter-julius-oketayot